sdk/storage/storage-file/review/storage-file.api.ts

```diff
@@ -0,0 +1,787 @@
+  getShareClient(shareName: string): ShareClient;
+  listShares(options?: ServiceListSharesOptions): PagedAsyncIterableIterator<Models.ShareItem, Models.ServiceListSharesSegmentResponse>;
+  setProperties(properties: Models.FileServiceProperties, options?: ServiceSetPropertiesOptions): Promise<Models.ServiceSetPropertiesResponse>;
+}
+
+// @public
+interface FileSetHTTPHeadersOptions extends FileAndDirectorySetPropertiesCommonOptions, CommonOptions {
+  abortSignal?: AbortSignalLike;
+}
+
+// @public
+interface FileSetMetadataOptions extends CommonOptions {
+  abortSignal?: AbortSignalLike;
+}
+
+// @public
+interface FileStartCopyOptions extends CommonOptions {
+  abortSignal?: AbortSignalLike;
+  metadata?: Metadata;
+}
+
+// @public
+class FileSystemAttributes {
+  archive: boolean;
+  directory: boolean;
+  hidden: boolean;
+  none: boolean;
+  noScrubData: boolean;
+  notContentIndexed: boolean;
+  offline: boolean;
+  static parse(fileAttributes: string): FileSystemAttributes;
+  readonly: boolean;
+  system: boolean;
+  temporary: boolean;
+  toString(): string;
+}
+
+// @public
+interface FileUploadRangeFromURLOptions extends Models.FileUploadRangeFromURLOptionalParams, CommonOptions {
+  abortSignal?: AbortSignalLike;
+}
+
+// @public
+interface FileUploadRangeOptions extends CommonOptions {
+  abortSignal?: AbortSignalLike;
+  contentMD5?: Uint8Array;
+  progress?: (progress: TransferProgressEvent) => void;
+}
+
+// @public
+export function generateAccountSASQueryParameters(accountSASSignatureValues: AccountSASSignatureValues, sharedKeyCredential: SharedKeyCredential): SASQueryParameters;
+
+// @public
+export function generateFileSASQueryParameters(fileSASSignatureValues: FileSASSignatureValues, sharedKeyCredential: SharedKeyCredential): SASQueryParameters;
+
+// @public
+class HttpHeaders {
+  constructor(rawHeaders?: RawHttpHeaders);
+  clone(): HttpHeaders;
+  contains(headerName: string): boolean;
+  get(headerName: string): string | undefined;
+  headerNames(): string[];
+  headersArray(): HttpHeader[];
+  headerValues(): string[];
+  rawHeaders(): RawHttpHeaders;
+  remove(headerName: string): boolean;
+  set(headerName: string, headerValue: string | number): void;
+  toJson(): RawHttpHeaders;
+  toString(): string;
+}
+
+// @public
+interface HttpOperationResponse extends HttpResponse {
+  blobBody?: Promise<Blob>;
+  bodyAsText?: string | null;
+  parsedBody?: any;
+  parsedHeaders?: {
+    [key: string]: any;
+  }
+  readableStreamBody?: NodeJS.ReadableStream;
+}
+
+// @public
+enum HttpPipelineLogLevel {
+  ERROR = 1,
+  INFO = 3,
+  OFF = 0,
+  WARNING = 2
+}
+
+// @public
+interface IHttpClient extends RequestPolicy {
+}
+
+// @public
+interface IHttpPipelineLogger {
+  log(logLevel: HttpPipelineLogLevel, message: string): void;
+  minimumLogLevel: HttpPipelineLogLevel;
+}
+
+// @public
+interface IPRange {
+  end?: string;
+  start: string;
+}
+
+// @public
+class LoggingPolicyFactory implements RequestPolicyFactory {
```

Comment:
Is it super important to have this? It seems like a lot of machinery to expose `logWarningIfTryOverThreshold` option. I would prefer to remove this if possible and add back in a way that doesn't require custom policy if needed/possible...

Reply:
I believe this is a tooling issue. `LoggingPolicyFactory` is not export in public api surface.

Reply:
Removed by @daviwil